File: backend/src/ResellerApi/Entities/Subscription.cs
using ResellerApi.Entities.Base;

namespace ResellerApi.Entities;

public class Subscription : BaseEntity
{
    public Guid CompanyId { get; set; }
    public Guid PlanId { get; set; }
    public string Status { get; set; } = null!; // TRIALING | ACTIVE | PAST_DUE | CANCELED | EXPIRED
    public string BillingCycle { get; set; } = "MONTHLY"; // MONTHLY | YEARLY
    public DateTime? TrialEndsAt { get; set; }
    public DateTime? CurrentPeriodStart { get; set; }
    public DateTime? CurrentPeriodEnd { get; set; }
    public DateTime? CanceledAt { get; set; }

    // Set when a checkout is started, consumed once the bKash callback confirms payment —
    // lets the callback know which plan/cycle the pending SubscriptionPayment is actually for.
    public Guid? PendingPlanId { get; set; }
    public string? PendingBillingCycle { get; set; }

    public Company Company { get; set; } = null!;
    public SubscriptionPlan Plan { get; set; } = null!;
    public ICollection<SubscriptionPayment> Payments { get; set; } = new List<SubscriptionPayment>();
}

File: frontend/src/types/subscription.ts
export interface SubscriptionPlan {
  id: string;
  code: string;
  name: string;
  staffSeatLimit: number;
  branchLimit: number;
  priceMonthly: number;
  priceYearly: number;
}

export interface SubscriptionStatus {
  status: "TRIALING" | "ACTIVE" | "PAST_DUE" | "CANCELED" | "EXPIRED";
  planCode: string;
  planName: string;
  billingCycle: "MONTHLY" | "YEARLY";
  trialEndsAt: string | null;
  currentPeriodEnd: string | null;
  staffSeatLimit: number;
  staffSeatsUsed: number;
  branchLimit: number;
  branchesUsed: number;
  isReadOnlyLocked: boolean;
}
